cmd/ci-operator/main.go

```diff
@@ -243,6 +258,38 @@ func main() {
 	opt.Report()
 }
 
+// setupLogger sets up logrus to print all logs to a file and user-friendly logs to stdout
+func setupLogger() (*secrets.DynamicCensor, io.Closer, error) {
+	logrusutil.ComponentInit()
+	logrus.SetLevel(logrus.TraceLevel)
+	censor := secrets.NewDynamicCensor()
+	logrus.SetFormatter(logrusutil.NewFormatterWithCensor(logrus.StandardLogger().Formatter, &censor))
+	logrus.SetOutput(ioutil.Discard)
+	logrus.AddHook(&writer.Hook{
```

Comment:
Why do we need to discard the output and use a hook when the formatter already uses the dynamic censor?

Reply:
This is the way that `logrus` wants it done lol, discard the default output and then hook it into the spots you want. If you didn't do this, the default output (to stdout) would go with Trace level and spam. This is unfortunately their architecture for this /shrug